src/mantle/mantle_query_sync.c

```diff
@@ -75,3 +76,133 @@ GR_RESULT grWaitForFences(
         return GR_ERROR_OUT_OF_MEMORY;
     }
 }
+
+
+GR_RESULT grGetFenceStatus(
+    GR_FENCE fence)
+{
+    GrFence* grFence = (GrFence*)fence;
+    if (fence == NULL){
+	return GR_ERROR_INVALID_POINTER;
+    }
+
+    VkResult res = vki.vkGetFenceStatus(grFence->device, grFence->fence);
+    switch (res){
```

Comment:
please add one space before the bracket for if/else/switch/etc, and make sure you use 4-space indentation everywhere